[943] Company cards displaying NaN and Infinite

CarlWiren90 opened the pull request:
### ✨ What’s Changed?

Added checks for 0, NaN and finite to avoid the company cards displaying NaN and -100% when no data is present. 
Now also treating a 0 value on latest reporting period as a No Data-point since it's the same logic we're using to determine NoData on the company details page.

### 📸 Screenshots (if applicable)

<!-- Add before/after images or UI previews -->
Before: 
<img width="1329" height="805" alt="image" src="https://github.com/user-attachments/assets/45b28a63-04e9-4e87-93cc-eab7088b8feb" />

After:
<img width="1379" height="905" alt="image" src="https://github.com/user-attachments/assets/75dc9c9b-f2a7-4522-802e-63eedd23790d" />


### 📋 Checklist

- [x] PR title starts with [#issue-number]; if no issue is applicable use: [fix], [feat], [prod], or [copy]
- [x] I've verified the change runs locally both on mobile and desktop
- [x] I've set the labels, issue, and milestone for the PR
- [ ] [OPTIONAL] I've created sub-tasks or follow-up issues as needed (check if NA)

### 🛠 Related Issue

Closes #943

CarlWiren90 commented:
I believe this now cover the cases where we were displaying weird numbers and are now correctly displaying No Data when there is none available. Also added the translation for "mdr" and "bn" to be displayed depending on SV/EN. Talked to @elvbom about this yesterday and the conditions inside the JSX is getting abit messy. Might be good to create a util to handle the logic for this and just return the "No Data" if the numbers don't meet the reqs. Perhaps we can merge this and that will be a follow up later on?

@kaylawoodbury @elvbom

elvbom commented on src/components/companies/list/CompanyCard.tsx:
```diff
@@ -52,14 +52,18 @@ export function CompanyCard({
   const previousEmissions = previousPeriod?.emissions?.calculatedTotalEmissions;
   const emissionsChange =
-    previousEmissions != null && currentEmissions != null
+    previousEmissions != null &&
+    currentEmissions != null &&
+    previousEmissions != 0 &&
+    currentEmissions != 0
```
This could maybe be divided into two consts with checks for previousEmissions and currentEmissions and then reused in more parts of the code?

elvbom reviewed (commented):
Good job! Some minor comments :)

CarlWiren90 commented:
Morning!

Separated the checks for previousEmissions and currentEmissions to two different consts as per your suggestion @elvbom. Therefore also removed the null checks inside the jsx. Added translation for "mdr" to toggle from "mdr" to "bn" depending on language. Also deleted commented out code inside the component.

elvbom commented on src/components/companies/list/CompanyCard.tsx:
```diff
@@ -234,34 +197,40 @@ export function CompanyCard({
+              {latestPeriodEconomyTurnover ? (
+                <Text variant="h6">
+                  {localizeUnit(
+                    latestPeriodEconomyTurnover.value / 1e9,
+                    currentLanguage,
+                  )}{" "}
+                  <span className="text-lg text-grey ml-1">
+                    {t("companies.card.turnoverAmount")}
                   </span>
-                )}
-              </Text>
-            </div>
-          )}
+                  <span className="text-lg text-grey ml-1">
+                    {latestPeriodEconomyTurnover.currency}
+                  </span>
+                  {turnoverAIGenerated && (
+                    <span className="ml-2">
+                      <AiIcon size="sm" />
+                    </span>
+                  )}
+                </Text>
+              ) : (
+                <Text variant="h6" className="text-grey">
+                  {t("companies.card.noData")}
+                </Text>
+              )}
+            </>
```
This could use a slight rewrite - it would need a check for latestPeriodEconomyTurnover.value and the <></> could be removed for example

CarlWiren90 replied:
So as to change line 209 to {latestPeriodEconomyTurnover?.value ? instead?

elvbom reviewed (commented):
Looking good! One minor comment :)
